tests/shared/src/test/scala/cats/tests/SyntaxSuite.scala

```diff
@@ -544,6 +545,70 @@ object SyntaxSuite {
     val nested: Nested[F, G, A] = fga.nested
   }
 
+  def testEither[F[_]: Applicative, A: Order, B: Order](): Unit = {
+    val either = mock[Either[A, B]]
+    val a = mock[A]
+    val b = mock[B]
+
+    val v1 = either.toValidated
+    val v2 = either.toValidatedNel
+    val v3 = either.toValidatedA[F]
+
+    val v4 = either.toEitherA
+    val v5 = either.toEitherT
+    val v6 = either.toEitherNel
+    val v7 = either.toEitherNec
+    val v8 = either.toEitherNes
+
+    val v9 = Either.left[A, B](a)
+    val v10 = Either.right[A, B](b)
+
+    val v11 = Either.leftNec(a)
+    val v12 = Either.rightNec(b)
+    val v13 = Either.leftNes(a)
+    val v14 = Either.rightNes(b)
+    val v15 = Either.leftNel(a)
+    val v16 = Either.rightNel(b)
+    val v17 = Either.leftA[F, A, B](a)
+  }
+
+  def testOption[F[_]: Applicative, G[_], A: Order, B: Order: Monoid](): Unit = {
+    val option = mock[Option[B]]
+    val a = mock[A]
+    val b = mock[B]
+    implicit val aega: ApplicativeError[G, B] = mock[ApplicativeError[G, B]]
+
+    val v1 = option.toInvalid(a)
+    val v2 = option.toInvalidNel(a)
+    val v3 = option.toInvalidA[F, A](a)
```

Comment:
This will result to `v3: Validated[F[A], A]`. I guess, the intent was to get `Validated[F[A], B]` instead. So although this test is technically correct, but a bit misleading. I think for tests like this one it should be helpful to specify each val's type explicitly. But again, it is not really that important, feel free to ignore.

Reply:
Actually it will be `Validated[F[B], A]`, but thanks, I will add types